andela/mrm_api
# CON-75 Send notification to admins when a room is under-booked

#### Description
This Pull request allows an administrator for the converge web app to receive notifications whenever a room is under-booked and underutilized to promote optimal space management.

#### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/mrm_api.git`
2. Setup project according to Readme.md
3. checkout to branch `story/CON-75-underbooked-rooms-notification`
4. run the following query to create an event. Make sure the eventId and other details are valid:

#### NB
- make sure the calendarId is correct.
- make sure the numberOfParticipants is less than the room capacity.

```
mutation {
  eventCheckin(
    eventId: "er0vj3v",
    calendarId: "xxxxxxxxx",
    numberOfParticipants: 1,
    eventTitle: "First LFA On-Boarding",
    startTime: "2018-08-14T06:00:00-07:00",
    endTime: "2018-08-14T07:00:00-07:00"
  ){
    event {
      id
      eventTitle
      room {
        id
        name
      }
    }
  }
}
```

5. To test the sockets, create an HTML file `index.html` and fill it with the following HTML code and change the location in the channel string to whatever location the cancelled event is happening, ie from `notificatons-lagos` to `notifications-kampala`

```
<!DOCTYPE html>
  <html lang="en">
  <head>
    <title>Flask_Chat_App</title>
    <script
      src="https://code.jquery.com/jquery-3.4.1.min.js"
      integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo="
      crossorigin="anonymous"></script>
  </head>
  <body>

    <h3 id="#messages" style='color: #ccc;font-size: 30px;'>No message yet..</h3>
    <div class="message_holder"></div>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/1.7.3/socket.io.min.js"></script>
    <script type="text/javascript">
      $(document).ready(function() {
        var socket = io.connect('http://127.0.0.1:5000');
        socket.on('connect', function() {
          socket.send('User has connected!');
        });
        socket.on('notifications-lagos', function(msg) {
          console.log(msg)
          console.log('Received message');
        });
      });
    </script>
  </body>
  </html>
```

#### Type of Change

#### How has this been tested

#### Checklist



#### JIRA
[CON-75](https://andela-apprenticeship.atlassian.net/jira/software/projects/CON/boards/15/backlog?label=backend&selectedIssue=CON-75)
